### ISSUE TITLE
linter: typescript/no-unnecessary-boolean-literal-compare extend detection

### ISSUE THREAD

**Issue by danciudev** [oxc-project/oxc#25706]
Hello, I noticed that the linter rule typescript/no-unnecessary-boolean-literal-compare doesn’t detect cases like the following example:

<img width="715" height="63" alt="Image" src="https://github.com/user-attachments/assets/f0500226-38e2-4040-8dd0-fe0156731c65" />

I’m not sure whether this should be considered a bug or whether the rule’s scope should be extended as a new feature.

**Comment by baevm** [comment]
if I understood your example right, i think this is already reported with `allowComparingNullableBooleansToTrue` option set to false (default is true) (https://oxc.rs/docs/guide/usage/linter/rules/typescript/no-unnecessary-boolean-literal-compare.html#configuration):

```
"typescript/no-unnecessary-boolean-literal-compare": ["error", {"allowComparingNullableBooleansToTrue": false}],
```

<img width="996" height="266" alt="Image" src="https://github.com/user-attachments/assets/1534d06e-5fa2-4c9d-871f-60924f29aed2" />
